File: api/integration-execution-contract/src/main/java/com/integration/execution/contract/model/RecordMetadata.java
package com.integration.execution.contract.model;

/**
 * Model for storing metadata about successfully processed records (added/updated)
 * Used in JSONB columns for detailed record tracking
 * Timestamp fields are nullable - null represents unset/unavailable timestamps
 */
public record RecordMetadata(
        String documentId,
        String title,
        String locationId,
        Long documentCreatedAt,
        Long documentUpdatedAt,
        Long locationCreatedAt,
        Long locationUpdatedAt
) {
}

File: web/src/api/models/RecordMetadata.ts
export interface RecordMetadata {
  documentId: string;
  title: string;
  locationId: string;
  documentCreatedAt: number | null;
  documentUpdatedAt: number | null;
  locationCreatedAt: number | null;
  locationUpdatedAt: number | null;
}

export interface FailedRecordMetadata extends RecordMetadata {
  errorMessage: string;
}
